examples/src/bin/embassy_usb_serial.rs

```diff
@@ -0,0 +1,130 @@
+//! CDC-ACM serial port example using embassy.
+//!
+//! This example should be built in release mode.
+
+//% CHIPS: esp32s2 esp32s3
+//% FEATURES: async embassy embassy-executor-thread embassy-time-timg0 embassy-generic-timers
+
+#![no_std]
+#![no_main]
+#![feature(type_alias_impl_trait)]
+
+use embassy_executor::Spawner;
+use embassy_futures::join::join;
+use embassy_usb::{
+    class::cdc_acm::{CdcAcmClass, State},
+    driver::EndpointError,
+    Builder,
+};
+use esp_backtrace as _;
+use esp_hal::{
+    clock::ClockControl,
+    embassy,
+    gpio::Io,
+    otg_fs::{
+        asynch::{Config, Driver},
+        Usb,
+    },
+    peripherals::Peripherals,
+    prelude::*,
+    system::SystemControl,
+    timer::TimerGroup,
+};
+
+#[main]
+async fn main(_spawner: Spawner) -> () {
+    esp_println::println!("Init!");
+    let peripherals = Peripherals::take();
+    let system = SystemControl::new(peripherals.SYSTEM);
+    let clocks = ClockControl::boot_defaults(system.clock_control).freeze();
+
+    embassy::init(&clocks, TimerGroup::new_async(peripherals.TIMG0, &clocks));
+    let io = Io::new(peripherals.GPIO, peripherals.IO_MUX);
+
+    let usb = Usb::new(peripherals.USB0, io.pins.gpio19, io.pins.gpio20);
+
+    // Create the driver, from the HAL.
+    let mut ep_out_buffer = [0u8; 1024];
+    let config = Config::default();
+
+    let driver = Driver::new(usb, &mut ep_out_buffer, config);
+
+    // Create embassy-usb Config
+    let mut config = embassy_usb::Config::new(0x16c0, 0x27dd);
```

Comment:
I think it would be more appropriate to use Espressif's VID, 12346 (0x303A hexadecimal).

We can also allocate a range of PIDs for esp-hal examples, just like we do for TinyUSB examples in the C SDK.

Reply:
Let's go with VID: `0x303A`, and PID: `0x3001`. The PID isn't yet finalized but we can update it once it is. Would you also mind updating the `usb_serial` example too?

Reply:
The PID 0x3001 has is now officially ours :)